EDU-164/EDU-864: update Search Attribute docs (#2203)
temporalio/documentation: docs-src/concepts/what-is-a-search-attribute.md

A Search Attribute is an indexed field used in a [List Filter](/concepts/what-is-a-list-filter) to filter a list of [Workflow Executions](/workflows#workflow-execution) that have the Search Attribute in their metadata.

Each Search Attribute is a key-value pair metadata object included in a Workflow Execution's visibility information.
This information is available in the Visibility store.

:::note

Search Attribute values are not encrypted because the Temporal Server must be able to read these values from the Visibility store when retrieving Workflow Execution details.

:::

Temporal provides some [default Search Attributes](#default-search-attributes), such as the `ExecutionStatus` of your Workflow Execution.

> You can also create [custom Search Attribute](#custom-search-attributes) keys in your Visibility store and assign values in a Workflow Execution.

Reviewer: ```suggestion
You can also create [custom Search Attribute](#custom-search-attributes) keys in your Visibility store and assign values when starting a Workflow Execution or in Workflow code.
```
Or during it

When using [Continue-As-New](/concepts/what-is-continue-as-new) or a [Temporal Cron Job](/concepts/what-is-a-temporal-cron-job), Search Attribute keys are carried over to the new Workflow Run by default.
Search Attribute values are only available for the duration of the Namespace's [Retention Period](/concepts/what-is-a-retention-period).

> Search Attributes are most effective for search purposes or tasks requiring collection-based resultsets.

Reviewer: ```suggestion
Search Attributes are most effective for search purposes or tasks requiring collection-based result sets.
```

> For business flow tasks, consider introducing a cache to capture the Workflow's internal state, or use a Query for entity-based state exposure.

Reviewer: ```suggestion
For business flow tasks in which you need to get information about a Workflow Execution, consider either
- storing state in a local variable and exposing it with a Query.
- storing state in an external data store via Activities and fetching it directly from the store.

For high throughput or low latency use cases, use the latter; however, note that the external data store may be slightly out of date (since time passes between the Workflow's state changing and the Activity updating the store).
```

A Search Attribute is an indexed field used in a [List Filter](/concepts/what-is-a-list-filter) to filter a list of [Workflow Executions](/workflows#workflow-execution) that have the Search Attribute in their metadata.

Each Search Attribute is a key-value pair metadata object included in a Workflow Execution's visibility information.
This information is available in the Visibility store.

:::note

Search Attribute values are not encrypted because the Temporal Server must be able to read these values from the Visibility store when retrieving Workflow Execution details.

:::

Temporal provides some [default Search Attributes](#default-search-attributes), such as the `ExecutionStatus` of your Workflow Execution.
You can also create [custom Search Attribute](#custom-search-attributes) keys in your Visibility store and assign values when starting a Workflow Execution or in Workflow code.

When using [Continue-As-New](/concepts/what-is-continue-as-new) or a [Temporal Cron Job](/concepts/what-is-a-temporal-cron-job), Search Attribute keys are carried over to the new Workflow Run by default.

> Search Attribute values are only available for the duration of the Namespace's [Retention Period](/concepts/what-is-a-retention-period).

Reviewer 2: ```suggestion
Search Attribute values are only available for as long as the workflow is.
```

I think that's a bit clearer. This makes it seem that even if a workflow is still running, the search attributes go away after a certain amount of time

> Search Attributes are most effective for search purposes or tasks requiring collection-based result sets.

Author: Paraphrasing the response I got in Slack from Mike: result sets are collections of state changes that can be reviewed for certain purposes (admin, ops, support).

> For business flow tasks in which you need to get information about a Workflow Execution, consider either

Author: Yeah, my bad. Will fix.
